akka-stream/src/main/scala/akka/stream/javadsl/Framing.scala

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright (C) 2015-2016 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.stream.javadsl
+
+import java.nio.ByteOrder
+
+import akka.NotUsed
+import akka.stream.scaladsl
+import akka.stream.stage._
+import akka.util.ByteString
+
+object Framing {
+
+  /**
+   * Creates a Flow that handles decoding a stream of unstructured byte chunks into a stream of frames where the
+   * incoming chunk stream uses a specific byte-sequence to mark frame boundaries.
+   *
+   * The decoded frames will not include the separator sequence.
+   *
+   * If there are buffered bytes (an incomplete frame) when the input stream finishes and ''allowTruncation'' is set to
+   * false then this Flow will fail the stream reporting a truncated frame.
+   *
+   * @param delimiter The byte sequence to be treated as the end of the frame.
+   * @param allowTruncation If turned on, then when the last frame being decoded contains no valid delimiter this Flow
```

Comment:
the opposite. If truncation is allowed, then it does not fail the stream if its, well, truncated :)